backend/store/etcd/organization_store.go

```diff
@@ -20,6 +20,55 @@ func getOrganizationsPath(name string) string {
 	return path.Join(EtcdRoot, organizationsPathPrefix, name)
 }
 
+// CreateOrganization creates an organization and a default environment with the
+// provided org
+func (s *Store) CreateOrganization(ctx context.Context, org *types.Organization) error {
+	if err := org.Validate(); err != nil {
+		return err
+	}
+
+	orgBytes, err := json.Marshal(org)
+	if err != nil {
+		return err
+	}
+
+	// Define a default environment for this new organization
+	env := types.Environment{
+		Name:         "default",
+		Description:  "Default organization",
+		Organization: org.Name,
+	}
+
+	envBytes, err := json.Marshal(env)
+	if err != nil {
+		return err
+	}
+
+	orgKey := getOrganizationsPath(org.Name)
+
+	res, err := s.client.Txn(ctx).
+		If(
+			// Ensure the organization does not already exist
```

Comment:
Why is it important to ensure that the organization does not already exist?

Reply:
To me, it's simply a question of UX; if I run `sensuctl organization create`, I don't expect it to update an organization. Same goes with `sensuctl organization update`, I wouldn't expect this command to create an organization if it's missing. Makes sense?

Reply:
Since Organizations are resources that are named by users and not Sensu, they are compatible with PUT semantics. If the store is designed this way, PUT semantics are impossible and we have to rely on POST and PATCH, each of them failing when the preconditions are not what the system expects.

What's nice about PUT is that it is idempotent - the operation has the same effect no matter how many times it is applied. This is really important for something like `sensuctl create -f`, where we create multiple resources, because we can repeat the operation without worry if one of the operations fails.

I think that for these types of resources, create and update should have identical semantics in the store. If an organization's description is replaced, nothing of value is lost. We only really need error-if-exists type semantics in the store in cases where creation has side effects, like destroying other bits of information that the system depends on.

We can still achieve the desired UX in sensuctl by using POST and PATCH - we simply have to attempt to load the resource first before calling `UpdateOrganization`. That's not transactional, but I think that's actually OK in these cases.

Reply:
You are totally right, it makes total sense. I'll make the required changes!

Reply:
Waiting on https://github.com/sensu/sensu-go/pull/1154 to be merged first!

Reply:
@echlebek Alright so I updated this PR so the new method `CreateOrganization` only creates an org if it does not already exist, while `UpdateOrganization` can create or update an org!